pkgs/by-name/ph/phase/package.nix

```
@@ -0,0 +1,102 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  unzip,
+  installShellFiles,
+  makeWrapper,
+  autoPatchelfHook,
+  zlib,
+  glibc,
+}:
+
+let
+  pname = "phase";
+  version = "1.18.6"; # This will be auto-updated by update.sh
+  hashes = lib.importJSON ./hashes.json;
+
+  # Binary target map
+  systemInfo = {
+    "x86_64-linux" = {
+      archivePath = "Linux-binary";
+      archString = "linux_amd64";
+    };
+    "aarch64-linux" = {
+      archivePath = "Linux-binary-arm64";
+      archString = "linux_arm64";
+    };
+    "x86_64-darwin" = {
+      archivePath = "Darwin-binary";
+      archString = "macos_amd64";
+    };
+    "aarch64-darwin" = {
+      archivePath = "Darwin-binary-arm64";
+      archString = "macos_arm64";
+    };
+  };
+
+  inherit (stdenv.hostPlatform) system;
+  info = systemInfo.${system} or (throw "Unsupported system: ${system}");
```

Unnecessary (package will refuse to evaluate when platform is not listed in `meta.platforms`)
```suggestion
  info = systemInfo.${system};
```